src/ir/possible-contents.cpp

```diff
@@ -0,0 +1,1628 @@
+                            ExpressionLocation{curr, 0}});
+    }
+  }
+
+  // Adds a result to the current function, such as from a return or the value
+  // that flows out.
+  void addResult(Expression* value) {
+    if (value && isRelevant(value->type)) {
+      for (Index i = 0; i < value->type.size(); i++) {
+        info.links.push_back(
+          {ExpressionLocation{value, i}, ResultLocation{getFunction(), i}});
+      }
+    }
+  }
+
+  void visitReturn(Return* curr) { addResult(curr->value); }
+
+  void visitFunction(Function* curr) {
+    // Vars have an initial value.
+    for (Index i = 0; i < curr->getNumLocals(); i++) {
+      if (curr->isVar(i)) {
+        Index j = 0;
+        for (auto t : curr->getLocalType(i)) {
+          if (t.isDefaultable()) {
+            info.links.push_back(
+              {getNullLocation(t), LocalLocation{curr, i, j}});
+          }
+          j++;
+        }
+      }
+    }
+
+    // Functions with a result can flow a value out from their body.
+    addResult(curr->body);
+
+    // See visitPop().
+    assert(handledPops == totalPops);
+  }
+
+  // Helpers
+
+  // Handles the value sent in a break instruction. Does not handle anything
+  // else like the condition etc.
+  void handleBreakValue(Expression* curr) {
+    BranchUtils::operateOnScopeNameUsesAndSentValues(
+      curr, [&](Name target, Expression* value) {
+        if (value && isRelevant(value->type)) {
+          for (Index i = 0; i < value->type.size(); i++) {
+            // Breaks send the contents of the break value to the branch target
+            // that the break goes to.
+            info.links.push_back(
+              {ExpressionLocation{value, i},
+               BreakTargetLocation{getFunction(), target, i}});
+          }
+        }
+      });
+  }
+
+  // Handles receiving values from breaks at the target (as in a block).
+  void handleBreakTarget(Expression* curr) {
+    if (isRelevant(curr->type)) {
+      BranchUtils::operateOnScopeNameDefs(curr, [&](Name target) {
+        for (Index i = 0; i < curr->type.size(); i++) {
+          info.links.push_back({BreakTargetLocation{getFunction(), target, i},
+                                ExpressionLocation{curr, i}});
+        }
+      });
+    }
+  }
+
+  // Connect a child's value to the parent, that is, all content in the child is
+  // now considered possible in the parent as well.
+  void receiveChildValue(Expression* child, Expression* parent) {
+    if (isRelevant(parent) && isRelevant(child)) {
+      // The tuple sizes must match (or, if not a tuple, the size should be 1 in
+      // both cases).
+      assert(child->type.size() == parent->type.size());
+      for (Index i = 0; i < child->type.size(); i++) {
+        info.links.push_back(
+          {ExpressionLocation{child, i}, ExpressionLocation{parent, i}});
+      }
+    }
+  }
+
+  // See the comment on CollectedFuncInfo::childParents.
+  void addSpecialChildParentLink(Expression* child, Expression* parent) {
```

Comment:
Is there a more descriptive word than "special" that we can use in this name? Alternatively, would dropping it and having `addChildParentLink` work? Or even `addParentLink`?